packages/demos-babylon/src/chatbotDemo.js

```diff
@@ -0,0 +1,182 @@
+    botAlias: 'Dev'
+  }
+  lex = new AwsFeatures.LexFeature(lexClient, botConfig);
+
+  initUi();
+  initConversationManagement();
+  acquireMicrophoneAccess();
+
+  return scene;
+}
+
+function initUi() {
+  // Set up interactions for UI buttons.
+  document.getElementById('startButton').onclick = () => startMainExperience();
+  document.getElementById('enableMicButton').onclick = () => acquireMicrophoneAccess();
+}
+
+/**
+ * Triggered when the user clicks the initial "start" button.
+ */
+function startMainExperience() {
+  showUiScreen('chatbotUiScreen');
+
+  // Speak a greeting to the user.
+  host.TextToSpeechFeature.play(
+    `Hello. How can I help?  You can say things like, "I'd like to rent a car," or, "Help me book a hotel".`
+  );
+}
+
+// ===== Chatbot functions =====
+
+let messageContainerEl;
+let transcriptTextEl;
+let lex;
+
+function initConversationManagement() {
+  // Use talk button events to start and stop recording.
+  const talkButton = document.getElementById('talkButton');
+  talkButton.onmousedown = () => lex.beginVoiceRecording();
+  talkButton.onmouseup = () => lex.endVoiceRecording();
+
+  // Use events dispatched by the LexFeature to present helpful user messages.
+  const { EVENTS } = AwsFeatures.LexFeature;
+  lex.listenTo(EVENTS.lexResponseReady, (response) => handleLexResponse(response));
+  lex.listenTo(EVENTS.recordBegin, () => hideUserMessages());
+  lex.listenTo(EVENTS.recordEnd, () => displayProcessingMessage());
+
+  // Create convenience references to DOM elements.
+  messageContainerEl = document.getElementById('userMessageContainer');
+  transcriptTextEl = document.getElementById('transcriptText');
+}
+
+/**
+ * Triggered whenever a response is received from the Lex chatbot.
+ * @param {object} response An object representing the Lex response. For a
+ * detailed description of this object's shape, see the documentation for the
+ * "data" callback argument described here:
+ * {@link https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/LexRuntime.html#postContent-property}
+ */
+function handleLexResponse(response) {
+  // Remove "processing" CSS class from message container.
+  messageContainerEl.classList.remove('processing');
+
+  // Display the user's speech input transcript.
+  displaySpeechInputTranscript(response.inputTranscript);
+
+  // Have the host speak the response from Lex.
+  host.TextToSpeechFeature.play(response.message);
+}
+
+function displaySpeechInputTranscript(text) {
+  transcriptTextEl.innerText = `“${text}”`;
+  messageContainerEl.classList.add('showingMessage');
+}
+
+function displayProcessingMessage() {
+  messageContainerEl.classList.add('processing');
+}
+
+function hideUserMessages() {
+  messageContainerEl.classList.remove('showingMessage');
+}
+
+/**
+ * Attempts to enable microphone access for Lex, triggering a browser permissions
+ * prompt if necessary.
+ */
+function acquireMicrophoneAccess() {
+  showUiScreen('micInitScreen');
+
+  lex.enableMicInput()
```

Comment:
Is there a reason to use the `.then` callback-style here instead of `await`?

Reply:
Yes. This is intentional. If I used "await" it would block the display of the scene while awaiting the user to allow mic access. I wanted the scene to continue rendering in the background while the prompt is displayed.

Reply:
Hmm. I may revisit this. I'm realizing now this was a remnant of a previous version where I needed to use .then(), but now that I've isolated it to it's own wrapper function I may be able to go back to `await`.